cmake/std/atdm/arm/environment.sh

```diff
@@ -0,0 +1,128 @@
+################################################################################
+#
+# Set up env on a ARM ATSE builds of Trilinos
+# This source script gets the settings from the ATDM_CONFIG_BUILD_NAME var.
+#
+################################################################################
+
+#
+# Deal with compiler versions
+#
+
+if [ "$ATDM_CONFIG_COMPILER" == "DEFAULT" ] ; then
+  export ATDM_CONFIG_COMPILER=GNU-7.2.0
+elif [[ "$ATDM_CONFIG_COMPILER" == "ARM-19.2"* ]]; then
+    export ATDM_CONFIG_COMPILER=ARM-19.2
+elif [[ "$ATDM_CONFIG_COMPILER" == "GNU"* ]]; then
+  if [[ "$ATDM_CONFIG_COMPILER" == "GNU" ]] ; then
+    export ATDM_CONFIG_COMPILER=GNU-7.2.0
+elif [[ "$ATDM_CONFIG_COMPILER" != "GNU-7.2.0" ]] ; then
+    echo
+    echo "***"
+    echo "*** ERROR: GNU COMPILER=$ATDM_CONFIG_COMPILER is not supported!"
+    echo "*** Only GNU compilers supported on this system are:"
+    echo "***   gnu (defaults to gnu-7.2.0)"
+    echo "***   gnu-7.2.0 (default)"
+    echo "***"
+    return
+  fi
+else
+  echo
+  echo "***"
+  echo "*** ERROR: COMPILER=$ATDM_CONFIG_COMPILER is not supported!"
+  echo "***"
+  return
+fi
+
+#
+# Allow KOKKOS_ARCH which is needed for CUDA builds
+#
+
+if [ "$ATDM_CONFIG_KOKKOS_ARCH" == "DEFAULT" ] ; then
+  unset ATDM_CONFIG_KOKKOS_ARCH
+fi
+
+echo "Using ARM ATSE compiler stack $ATDM_CONFIG_COMPILER to build $ATDM_CONFIG_BUILD_TYPE code with Kokkos node type $ATDM_CONFIG_NODE_TYPE"
+
+export ATDM_CONFIG_USE_NINJA=ON
+
+# Get ATDM_CONFIG_NUM_CORES_ON_MACHINE for this machine
+source $ATDM_SCRIPT_DIR/utils/get_num_cores_on_machine.sh
+
+if [ "$ATDM_CONFIG_NUM_CORES_ON_MACHINE" -gt "20" ] ; then
+  export ATDM_CONFIG_MAX_NUM_CORES_TO_USE=20
+  # NOTE: We get links crashing if we try to use to many processes.  ToDo: We
+  # should limit the number of processes that ninja uses to link instead of
+  # reducing the overrall parallel build level like this.
+else
+  export ATDM_CONFIG_MAX_NUM_CORES_TO_USE=$ATDM_CONFIG_NUM_CORES_ON_MACHINE
+fi
+
+export ATDM_CONFIG_BUILD_COUNT=$ATDM_CONFIG_MAX_NUM_CORES_TO_USE
+# NOTE: Use as many build processes and there are cores by default.
+
+module purge
```

Comment:
It seems like you would want to do this earlier, to avoid clobbering any of the above environment variables (just in case), but it shouldn't matter too much, I suppose.